### ISSUE TITLE
LinkExtension.AddLink

### ISSUE THREAD

**Issue by adambajguz** [higtrollers/Sciensoft.Hateoas#19]
In my oppinion LinkExtension.AddLink does not have enough flexibility. It is not possible to set custom JSON serializer options, use other methods instead of AddMvcCore like AddControllers. This may lead to braking the existing code. 

The solution could be to extract parts of LinkExtension.AddLink e.g.

```C#
services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
services.AddTransient<HateoasUriProvider<InMemoryPolicyRepository.SelfPolicy>, HateoasSelfUriProvider>();
services.AddTransient<HateoasUriProvider<InMemoryPolicyRepository.RoutePolicy>, HateoasRouteUriProvider>();
services.AddTransient<HateoasUriProvider<InMemoryPolicyRepository.CustomPolicy>, HateoasCustomUriProvider>();
services.AddTransient<HateoasUriProvider<InMemoryPolicyRepository.ExternalPolicy>, HateoasExternalUriProvider>();

services.AddTransient<IHateoasResultProvider, HateoasResultProvider>();
```

so that they can be executed in user code without calling services.AddMvcCore implicitly in LinkExtension.AddLink.

Other solution may be making the classes public.

**Comment by AlexzSouz** [comment]
Hello @adambajguz , I'm looking into a better way to have custom JSON serialization. However, I believe this issue would be related to #18 in a way, and I'll notify you once I have some update on this one. Thanks!

**Comment by AlexzSouz** [comment]
@adambajguz I've registered the filters using options configuration instead of `AddMvcCore` to avoid breaking consumers code. It will be available in the release 3.1.1.

I will close this issue but feel free to re-open it if you have any concern.
